dc-commons-xml/src/main/java/de/digitalcollections/commons/xml/xpath/DocumentReader.java

```diff
@@ -26,30 +26,82 @@
   private final List<String> rootPaths;
   private final XPathWrapper xpw;
 
+  private final VariableResolver variableResolver;
+
   public DocumentReader(Document doc, List<String> rootPaths, String namespace) {
     this.rootPaths = rootPaths;
     this.xpw = new XPathWrapper(doc);
     if (namespace != null) {
       xpw.setDefaultNamespace(namespace);
     }
+
+    variableResolver = new VariableResolver(rootPaths, xpw);
   }
 
   public Document getDocument() {
     return xpw.getDocument();
   }
 
   public String readValue(List<String> expressions) throws XPathMappingException {
-    return resolveVariable(expressions.toArray(new String[] {})).values().stream()
+    return variableResolver
```

Comment:
wahrscheinlich bin ich nicht genug in der materie, aber ich versteh es nicht: was sind Variablen, was Expressions, ... Javadoc wäre nötig

Reply:
Javadoc wurde nachgezogen